### PR TITLE
Rapid7 playbooks

### PR DESCRIPTION
**Pull request by vu-socprime**
Added Rapid7 Custom API Logic App Connector and 3 playbooks

### CODE REVIEW

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    * Playbook Name: Enter the playbook name here
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichVulnerabilityInfo%2Fazuredeploy.json) [![Deploy to Azure](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichVulnerabilityInfo%2Fazuredeploy.json)
+
+### Post-Deployment instructions
+
+#### a. Authorize connections
+
+Once deployment is complete, authorize each connection.
+
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md:33]: Steps missing to setup Rapid 7 custom connector.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-RunScan%2Fazuredeploy.json) [![Deploy to Azure](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-RunScan%2Fazuredeploy.json)
+
+### Post-Deployment instructions
+
+#### a. Authorize connections
+
+Once deployment is complete, authorize each connection.
+
+1. Click the Azure Sentinel connection resource
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:38]: Name of all the Sentinel/Teams connection are very generic. Update the name to be very specific and update the playbook trigger name here.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for other connections
+
+#### b. Configurations in Sentinel
+
+1. In Azure sentinel, analytical rules should be configured to trigger an incident that contains IPs. In the *Entity maping* section of the analytics rule creation workflow, IP should be mapped to **Address** identitfier of the **IP** entity type. Check the [documentation](https://docs.microsoft.com/azure/sentinel/map-data-fields-to-entities) to learn more about mapping entities.
+2. Configure the automation rules to trigger the playbook.
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:48]: Please add the required link here.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+<img src="./teams_screenshot.png" width="50%"/><br>
+
+5. Runs scans for selected IPs using chosen scan engines.
+6. Add inforamtions about launched scans as a comment to the incident.
+
+<img src="./playbook_screenshot.png" width="50%"/><br>
+
+### Prerequisites
+
+1. Prior to the deployment of this playbook, Rapid7 InsightVM API Connector needs to be deployed under the same subscription.
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:21]: Add the required link here.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+2. Obtain Rapid7 InsightVM API credentials. Refer to Rapid7 InsightVM API Custom Connector documentation.
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:22]: Add the required link here.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+When a new sentinel incident is created, this playbook gets triggered and performs the following actions:
+
+1. Obtains IPs from the incident.
+2. Searches asset ids by the IPs.
+3. Gets vulnerability ids.
+4. Gets vulnerability information.
+5. Adds obtained information as a comment to the incident.
+
+<img src="./playbook_screenshot.png" width="50%"/><br>
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md:13]: Update the screenshot with the latest copy. Action names in the image do not match with what gets deployed.

File: Solutions/Rapid7InsightVM/Playbooks/readme.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
+# Rapid7 InsightVM Logic Apps connector and playbook templates
+
+<img src="./logo.png" alt="drawing" width="20%"/><br>
+
+## Table of Contents
+
+1. [Overview](#overview)
+1. [Custom Connectors + 3 Playbook templates deployment](#deployall)
+1. [Authentication](#importantnotes)
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/readme.md:9]: Incorrect mapping of the link.

File: Solutions/Rapid7InsightVM/Playbooks/readme.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+5. Click Save
+6. Repeat steps for other connections
+
+#### b. Configurations in Sentinel
+
+Each Playbook requires a different type of configuration. Check documentation for each Playbook.
+
+<a name="limitations">
+
+## Known Issues and Limitations
```
Comment (devikamehra) [inline Solutions/Rapid7InsightVM/Playbooks/readme.md:94]: Since this is empty, can we remove it for now.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Please make the required changes.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+#### a. Authorize connections
+
+Once deployment is complete, authorize each connection.
+
+1. Click the Azure Sentinel connection resource
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for other connections
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md:38]: It is instructions how to authorize API connections for playbook, there is no need to mention action names as each API connection authorises all the actions in the playbook.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+#### b. Configurations in Sentinel
+
+1. In Azure sentinel, analytical rules should be configured to trigger an incident that contains IPs. In the *Entity maping* section of the analytics rule creation workflow, IP should be mapped to **Address** identitfier of the **IP** entity type. Check the [documentation](https://docs.microsoft.com/azure/sentinel/map-data-fields-to-entities) to learn more about mapping entities.
+2. Configure the automation rules to trigger the playbook.
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md]: done

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+1. Obtains IPs from the incident.
+2. Searches asset ids by the IPs.
+3. Gets assets information.
+4. Adds obtained information as a comment to the incident.
+
+<img src="./playbook_screenshot.png" width="50%"/><br>
+
+### Prerequisites
+
+1. Prior to the deployment of this playbook, Rapid7 InsightVM API Connector needs to be deployed under the same subscription.
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md]: updated

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+2. Obtain Rapid7 InsightVM API credentials. Refer to Rapid7 InsightVM API Custom Connector documentation.
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md]: updated

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichVulnerabilityInfo%2Fazuredeploy.json) [![Deploy to Azure](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichVulnerabilityInfo%2Fazuredeploy.json)
+
+### Post-Deployment instructions
+
+#### a. Authorize connections
+
+Once deployment is complete, authorize each connection.
+
+1. Click the Azure Sentinel connection resource
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md]: There is no need to mention action names as each API connection authorises all the actions in the playbook.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+2. Click edit API connection
+3. Click Authorize
+4. Sign in
+5. Click Save
+6. Repeat steps for other connections
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichVulnerabilityInfo/readme.md:39]: `6. Repeat steps for other connections` means that each connection in "API connections" section should be authorised the same way.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+### Deployment instructions
+
+1. To deploy the Playbook, click the Deploy to Azure button. This will launch the ARM Template deployment wizard.
+2. Fill in the required paramteres:
+    * Playbook Name: Enter the playbook name here
+
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:32]: Parameters added. But I could not find the official documentation about getting these ids.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md (hunk 3 above)
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md]: done

File: Solutions/Rapid7InsightVM/Playbooks/readme.md (hunk 7 above)
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/readme.md:9]: fixed

File: Solutions/Rapid7InsightVM/Playbooks/readme.md (hunk 8 above)
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/readme.md]: updated

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    * Playbook Name: Enter the playbook name here
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichIncidentWithAssetInfo%2Fazuredeploy.json) [![Deploy to Azure](https://aka.ms/deploytoazuregovbutton)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FSolutions%2FRapid7InsightVM%2FPlaybooks%2FPlaybooks%2FRapid7InsightVM-EnrichIncidentWithAssetInfo%2Fazuredeploy.json)
+
+### Post-Deployment instructions
+
+#### a. Authorize connections
+
+Once deployment is complete, authorize each connection.
+
```
[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md:32]: Can we explicitly mention this step as Rapid7 connection is custom made for our solution?

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:32]: If there is no official document, can you lay down the steps as to how you got these details?

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+2. Searches asset ids by the IPs.
+3. Obtains a list of scan engines.
+4. Sends an adaptive card to the Teams channel where the user can choose an action to be taken.
+
+<img src="./teams_screenshot.png" width="50%"/><br>
+
+5. Runs scans for selected IPs using chosen scan engines.
+6. Add inforamtions about launched scans as a comment to the incident.
+
+<img src="./playbook_screenshot.png" width="50%"/><br>
```
[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:17]: Names are different before we authorize. Please consider this resolved.

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-RunScan/readme.md:32]: Understood

File: Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/Rapid7InsightVM/Playbooks/Playbooks/Rapid7InsightVM-EnrichIncidentWithAssetInfo/readme.md:32]: Yes. I updated readme files.

**Comment by vmanojreddy** [comment]
@devikamehra Please take a look on resolve comments and provide your sign off. thanks!!!

**Review by devikamehra — APPROVED** [review]

**Review by anki-narravula — CHANGES_REQUESTED** [review]
Hi @vu-socprime - Can you please make following changes. Thanks in advance

1. For all playbooks, metadata object is missing. With the new template spec (for gallery), this is mandatory. Please look at the [ latest guide for playbooks contribution](https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide)
2. For all playbooks custom connector name must be included in the Parameters and use the parameter directly in "id" attribute of API. For example 
```
 "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', variables('customApis_vendorproduct_name'))]"
 "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', parameter('customApis_rapid7'))]"
```
3. In CustomConnector - name of the custom connector must be a parameter and keep default value (same as the current)

**Comment by vu-socprime** [comment]
> Hi @vu-socprime - Can you please make following changes. Thanks in advance
> 
> 1. For all playbooks, metadata object is missing. With the new template spec (for gallery), this is mandatory. Please look at the [ latest guide for playbooks contribution](https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide)
> 2. For all playbooks custom connector name must be included in the Parameters and use the parameter directly in "id" attribute of API. For example
> 
> ```
>  "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', variables('customApis_vendorproduct_name'))]"
>  "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Web/customApis/', parameter('customApis_rapid7'))]"
> ```
> 
>  3. In CustomConnector - name of the custom connector must be a parameter and keep default value (same as the current) also host name (API host) can be parameterized
> 4. Wherever sentinel connection used, use ManagedIdentity always - [refer template for an example](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json)

Hi @anki-narravula 
The playbooks were updated according to your feedback.

**Review by anki-narravula — APPROVED** [review]
Looks good. Approved.

**Pull request merged**